runtime/src/append_vec.rs

```diff
@@ -174,14 +208,27 @@ impl AppendVec {
 
     #[allow(clippy::mutex_atomic)]
     pub fn set_file<P: AsRef<Path>>(&mut self, path: P) -> io::Result<()> {
-        self.path = path.as_ref().to_path_buf();
         let data = OpenOptions::new()
             .read(true)
             .write(true)
             .create(false)
             .open(&path)?;
 
+        // this AppendVec must not hold actual file;
+        assert_eq!(self.file_size, 0);
+
+        let current_len = self.current_len.load(Ordering::Relaxed);
+        assert_eq!(current_len, *self.append_offset.lock().unwrap());
+
+        let file_size = std::fs::metadata(&path)?.len();
+        assert!(current_len <= file_size as usize);
+
+        AppendVec::sanitize_mmap_size(file_size as usize)?;
+
         let map = unsafe { MmapMut::map_mut(&data)? };
+
+        self.file_size = file_size;
```

Comment:
Update `file_size` correctly, will be used later PRs to check the sum of contained acconts doesn't exceed this and moreover exactly matches to the `current_len`.

Reply:
> Update `file_size` correctly, will be used later PRs to check the sum of contained acconts doesn't exceed this

As promised by 2 weeks ago's myself, this is done by [this](https://github.com/solana-labs/solana/pull/7464/files#diff-657d1e7648a9fb74cb1582a9c48beb4bR232).